exercise/templates/exercise/exercise.html

```diff
@@ -25,6 +25,9 @@
         {% endif %}
     {% endif %}
 >
+  {% if latest_enrollment_submission %}
+    {{ latest_enrollment_submission.submission_data|json_script:"enroll-submission-data" }}
```

Comment:
This exposes all the fields from the previous form. Do we have anyway to limit the fields? For example, could you read the form_spec and only include the fields which have the same key?

Reply:
What is the danger in exposing all the fields? It is the same user that made the submission.

Reply:
Working toward a place where teacher of a course can't access students data from another course. Currently, this is of course possible via the API, which is a huge security issue :(

Reply:
As a minimum, add a comment with `TODO: fix leaking of personal details by limiting amount of informations exposed here` or so here, so this problematic part of a code can be found later on.